File: C++/205. Isomorphic Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char,int> ms;
        unordered_map<char,int> mt;
        for (int i = 0; i < s.size(); i++) {
            if (ms[s[i]] != mt[t[i]]){
                return false;
            }
            ms[s[i]] = i + 1;
            mt[t[i]] = i + 1;
        }
        return true;
    }
};

File: C#/205. Isomorphic Strings.cs
public class Solution {
    public bool IsIsomorphic(string s, string t) {
        var ms = new Dictionary<char, int>();
        var mt = new Dictionary<char, int>();
        for (int i = 0; i < s.Length; i++) {
            if (!ms.ContainsKey(s[i])) {
			    ms.Add(s[i], t[i]);
		    }
		    if (!mt.ContainsKey(t[i])) {
		    	mt.Add(t[i], s[i]);
		    }
            if (ms[s[i]] != t[i] || mt[t[i]] != s[i]) {
			    return false;
		    }
        }
        return true;
    }
}
